### ISSUE TITLE
Change the UserAgent to show the application name

### ISSUE THREAD

**Issue by lleirborras** [ifad/colore-client#3]
As we have in many other clients (if not all) we should be able to change the user agent or even to force the user agent to show the application name instead of the default `rest-client/X.Y.Z`.

**Comment by vjt** [comment]
Agreed. Relevant code in Hawk: https://github.com/ifad/hawk/blob/26dcd3e07f7e1ceee15959ef1997d419432c4608/lib/hawk/model/connection.rb#L73

**Comment by lleirborras** [comment]
Unfortunately `RestClient` is not ready
https://github.com/rest-client/rest-client/blob/51032140c9de747520278f1834156134944eb61b/lib/restclient/request.rb#L631
https://github.com/rest-client/rest-client/blob/c469b3629ca826dac77416fc39adce190cde2a82/lib/restclient/platform.rb#L45
